pkg/snapstore/abs_snapstore.go

```diff
@@ -116,6 +115,23 @@ func getCredentials(prefixString string) (string, string, error) {
 		return credentials.StorageAccount, credentials.SecretKey, nil
 	}
 
+	// TODO: @renormalize Remove this extra handling in v0.31.0
+	// Check if a JSON file is present in the directory, if it is present -> the JSON file must be used for credentials.
+	if dir, isSet := os.LookupEnv(prefixString + absCredentialDirectory); isSet {
+		jsonCredentialFile, err := findFileWithExtensionInDir(dir, ".json")
+		if err != nil {
+			return "", "", fmt.Errorf("error while finding a JSON credential file in %v directory with error: %w", dir, err)
+		}
+		if jsonCredentialFile != "" {
+			credentials, err := readABSCredentialsJSON(jsonCredentialFile)
+			if err != nil {
+				return "", "", fmt.Errorf("error getting credentials using %v JSON file in a directory with error: %w", jsonCredentialFile, err)
+			}
+			return credentials.StorageAccount, credentials.SecretKey, nil
+		}
+		// Non JSON credential files might exist in the credential directory, do not return
+	}
+
 	if dir, isSet := os.LookupEnv(prefixString + absCredentialDirectory); isSet {
```

Comment:
Can I know why can't we just club these two `if` statements together, since then there would be no code redundancy and also it will be easy to understand the flow of fetching the credentials. As the flow goes like 

```
See if `absCredentialDirectory` is set, if yes -> Check if `jsonCredentialFile` is present, if yes, read from there, if no -> proceed to read the credentials from individual files
```

I presume you wanted to keep it this way since it's a temporary change and will be reverted back in a few releases but I think 3 releases is long enough time considering there are security issues and some major upgrades going on due to which the releases won't be as frequent. I believe even if it's temporary, it will be a bit easier for the users to understand the change in a neat flow. WDYT?

Reply:
I don't have a strong preference for either, but I feel this coding style is easier to read. Seperating this handling into a different `if` makes the intent clear that this handling is sepearate from the other two, i.e. an edge case that is being handled. If you strongly feel clubbing makes the code look cleaner, I will make the necessary changes.

Reply:
I don't have a strong opinion on this, but I personally feel that this is easier to read, than having a huge blob in an `if`. A separate `if` for this case makes the intent clear that the case being handled is different from the other two, because it is different from what is typically understood of how credentials are sent to etcd-backup-restore.

If you have a strong preference for this code in the same `if` condition, I'll make the changes accordingly.

Reply:
Since I don't have strong opinion on this and the below comment, I leave it to you to decide. You can close or address this depending on your opinion when you come to decide on this section of code since Ishan also has some reviews on this piece of code.

Reply:
I've expressed the same opinion with Ishan as well and he's okay with it. I'll be keeping this section as is. Thanks for the review!